File: app/src/main/java/ua/hospes/rtm/domain/race/models/PitStopAssign.java
package ua.hospes.rtm.domain.race.models;

/**
 * @author Andrew Khloponin
 */
public enum PitStopAssign {
    MANUAL,
    PREVIOUS,
    NEXT;


    public static PitStopAssign fromString(String string) {
        for (PitStopAssign pitStop : values())
            if (pitStop.name().equalsIgnoreCase(string)) return pitStop;
        return MANUAL;
    }
}

File: app/src/main/java/ua/hospes/rtm/domain/race/models/PitStopAssign.kt
package ua.hospes.rtm.domain.race.models

/**
 * @author Andrew Khloponin
 */
enum class PitStopAssign {
    MANUAL,
    PREVIOUS,
    NEXT;


    companion object {


        fun fromString(string: String): PitStopAssign {
            for (pitStop in values())
                if (pitStop.name.equals(string, ignoreCase = true)) return pitStop
            return MANUAL
        }
    }
}